src/daft-core/src/series/ops/agg.rs

```diff
@@ -61,6 +62,21 @@ impl Series {
                 .into_series()),
                 None => Ok(DaftSumAggable::sum(&self.downcast::<Float64Array>()?)?.into_series()),
             },
+            Decimal128(_, _) => match groups {
+                Some(groups) => Ok(Decimal128Array::new(
+                    self.field().clone(),
+                    DaftSumAggable::grouped_sum(
+                        &self.as_physical()?.downcast::<Int128Array>()?,
+                        groups,
+                    )?,
+                )
+                .into_series()),
+                None => Ok(Decimal128Array::new(
+                    self.field().clone(),
```

Comment:
When you are summing over decimal values, the precision needed to encode those values can change. So instead of keeping the same field with the same dtype, I would suggest setting the new precision to min(38, current precision + 19).

38 because that is the max precision of a Decimal128, 19 because that is the most the precision can increase by assuming that there are at most 2^64 rows of numbers (an assumption also made in other parts of Daft)

Reply:
Make sure also to change `try_sum_supertype` to update the same way